sdc/tests/test_series.py

```diff
@@ -6252,6 +6189,482 @@ def test_impl(A, B):
                 expected_msg = 'Operator mul(). Not supported between operands of types:'
                 self.assertIn(expected_msg, str(raises.exception))
 
+    @skip_sdc_jit("StringArray reflection was not implemented in old-pipeline")
+    @skip_numba_jit("TODO: support StringArray reflection")
+    def test_series_setitem_str_reflection(self):
+        """ Verifies that changes made to string Series passed into a jitted function
+            are propagated back to the native python object.
+        """
+        def test_impl(S, idx, val):
+            S[idx] = val
+            return S
+        hpat_func = self.jit(test_impl)
+
+        S = pd.Series(['cat', '', 'bbb', '', 'a', None, 'a', '', None, 'b'])
+        idx, value = 0, 'dog'
+        result = hpat_func(S, idx, value)
+        pd.testing.assert_series_equal(result, S)
+
+    def _test_series_setitem(self, all_data, indexes, idxs, values, dtype=None):
+        """ Common function used by setitem tests to compile and run setitem on provided data"""
+        def test_impl(A, i, value):
+            A[i] = value
+        hpat_func = self.jit(test_impl)
+
+        for series_data in all_data:
+            for series_index in indexes:
+                S = pd.Series(series_data, series_index, dtype=dtype)
+                for idx in idxs:
+                    for value in values:
```

Comment:
You could use `itertools.product` to combine the loops in the future. Now we can skip this.

Reply:
Will be done in #582